java/code/src/com/redhat/rhn/domain/contentmgmt/ContentManager.java

```diff
@@ -0,0 +1,229 @@
+                .orElseThrow(() -> new ContentManagementException("Content Project with label " + projectLabel +
+                        " not found."));
+    }
+
+    /**
+     * Look up Content Environment based on its label, Content Project label and Org
+     *
+     * @param envLabel - the Content Environment label
+     * @param contentLabel - the Content Project label
+     * @param user - the user
+     * @return the optional of matching Content Environment
+     */
+    public static Optional<ContentEnvironment> lookupEnvironment(String envLabel, String contentLabel, User user) {
+        return lookupProject(contentLabel, user)
+                .flatMap(cp -> ContentProjectFactory.lookupEnvironmentByLabelAndProject(envLabel, cp));
+    }
+
+    /**
+     * Update Content Environment
+     *
+     * @param envLabel - the Environment label
+     * @param projectLabel - the Content Project label
+     * @param newName - new name
+     * @param newDescription - new description
+     * @param user - the user
+     * @throws ContentManagementException - if the Environment is not found
+     * @throws PermissionException if given user does not have required role
+     * @return the updated Environment
+     */
+    public static ContentEnvironment updateEnvironment(String envLabel, String projectLabel, Optional<String> newName,
+            Optional<String> newDescription, User user) {
+        ensureOrgAdmin(user);
+        return lookupEnvironment(envLabel, projectLabel, user)
+                .map(env -> {
+                    newName.ifPresent(name -> env.setName(name));
+                    newDescription.ifPresent(desc -> env.setDescription(desc));
+                    return env;
+                })
+                .orElseThrow(() -> new ContentManagementException("Environment with label " + envLabel +
+                        " in Project " + projectLabel + " not found."));
+    }
+
+
+    /**
+     * Remove a Content Environment
+     *
+     * @param envLabel - the Content Environment label
+     * @param projectLabel - the Content Project label
+     * @param user - the user
+     * @throws PermissionException if given user does not have required role
+     * @return number of deleted objects
+     */
+    public static int removeEnvironment(String envLabel, String projectLabel, User user) {
+        ensureOrgAdmin(user);
+        return lookupEnvironment(envLabel, projectLabel, user)
+                .map((env) -> {
+                    ContentProjectFactory.removeEnvironment(env);
+                    return 1;
+                })
+                .orElse(0);
+    }
+
+    /**
+     * Ensures that given user has the Org admin role
+     *
+     * @param user the user
+     * @throws com.redhat.rhn.common.security.PermissionException if the user does not have Org admin role
+     */
+    private static void ensureOrgAdmin(User user) {
+        if (!user.hasRole(ORG_ADMIN)) {
+            throw new PermissionException(ORG_ADMIN);
```

Comment:
Why not `throwing` [this other](https://github.com/SUSE/spacewalk/blob/Manager-3.2/java/code/src/com/redhat/rhn/frontend/xmlrpc/BaseHandler.java#L300) `Permission` which looks more specific?

Note: I am not arguing this is not correct, I just want to be sure we are choosing the proper `Exception` in line with the product.

Reply:
The one you mentioned is from the `xmlrpc` package. I prefer not to use `xmlrpc` parts from the `domain`/`manager` packages. `ContentManager` is used in the frontend code as well - it could be a bit confusing, when a Web UI request causes an xmlrpc exception.

Reply:
HA!! Good point, I overlooked the `xmlrpc` thing. Generally speaking, it would be good to have just `one single exception type` for each functionality, but this is not the case nor the pattern used in the whole project. That's fine then. Thanks for pointing that out. :+1: